packages/flutter_tools/lib/src/linux/linux_doctor.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/io.dart';
+import '../base/process_manager.dart';
+import '../doctor.dart';
+
+/// A validator that checks for GCC and the correct gtk version.
+class LinuxDoctorValidator extends DoctorValidator {
+  const LinuxDoctorValidator() : super('Linux toolchain - develop for Linux desktop applications');
+
+  @override
+  Future<ValidationResult> validate() async {
+    ValidationType validationType = ValidationType.installed;
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    /// Check for some version of GCC.
+    final ProcessResult gccResult = await processManager.run(const <String>[
+      'g++',
+      '--version',
+    ]);
+    if (gccResult.exitCode != 0) {
+      validationType = ValidationType.missing;
+      messages.add(ValidationMessage.error('GCC executable missing'));
+    } else {
+      messages.add(ValidationMessage(gccResult.stdout));
+    }
+
+    /// Check for the correct version of gtk.
+    final ProcessResult gtkResult = await processManager.run(const <String>[
+      'pkg-config',
```

Comment:
We'd need to make sure pkg-config was present first. But as with GCC, I don't think we should actually be requiring that. The shell does have a runtime dependency on GTK, but that could be satisfied without pkg-config being present.

Reply:
These need not be required, but it could be a hint. Is there a better method to detect whether gtk is there at runtime?

Reply:
To be clear, when I say it's a runtime dependency I mean that *end users* of the built application need GTK; it's not a developer-specific thing. That's why I don't think `doctor` should check for it.

> Is there a better method to detect whether gtk is there

It's Linux, so as always the answer seems to be "it depends on your distribution!"

Honestly I think that without a bunch of fallback logic and iteration with different distros we're likely to create a lot of false-positives (i.e., whatever we use to check of the presence of GTK is going to be missing more often than GTK), and that we're better off not doing this for now, and seeing if there's really a need for a GTK check in practice based on feedback.